shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/ApplicationContext.java

```diff
@@ -53,16 +55,39 @@ public static <T> Set<Class<? extends T>> getImplementors(Class<T> interfaze) {
     }
 
     public static void injectRegisteredImplementations() {
-        injectRegisteredImplementations(false);
+        doInjectRegisteredImplementations();
     }
 
-    protected static void injectRegisteredImplementations(boolean skipMissingImplementations) {
+    protected static void doInjectRegisteredImplementations() {
         var fields = Reflection.getFieldsAnnotatedWith(Inject.class);
 
-        fields.forEach(field -> injectIntoField(field, skipMissingImplementations));
+        fields.forEach(ApplicationContext::injectIntoField);
+    }
+
+    public static void injectIntoNonSingleton(Object instance) {
+        var fields = Reflection.getFieldsAnnotatedWithInstance(instance.getClass(), Inject.class);
+
+        fields.forEach(field -> injectIntoField(field, instance));
+    }
+
+    private static void injectIntoField(Field field, Object instance) {
```

Comment:
There is a lot of code duplication between this `injectIntoField` method and the original `injectIntoField` method.  I bet the original method could be refactored to call the new one.

Reply:
Good catch. I will refactor it.